site/src/pages/TemplatePage/TemplateEmbedPage/TemplateEmbedPage.tsx

```diff
@@ -55,11 +59,16 @@ function getClipboardCopyContent(
 	const deploymentUrl = `${window.location.protocol}//${window.location.host}`;
 	const createWorkspaceUrl = `${deploymentUrl}/templates/${organization}/${templateName}/workspace`;
 	const createWorkspaceParams = new URLSearchParams(buttonValues);
+	if (createWorkspaceParams.get("name") === "") {
+		createWorkspaceParams.delete("name"); // skip default workspace name if undefined
```

Comment:
```suggestion
		createWorkspaceParams.delete("name"); // no default workspace name if empty
```

we are turning it into `undefined` here by deleting it. maybe that's what you meant? but undefined isn't the condition here, it's the desired end value

Reply:
I suppose I owe you an explanation. The `delete` operation is there to preserve backward compatibility with URLs that don’t include name. The reason an empty `name` shows up is because it’s initialized here:

```ts
const buttonValues: ButtonValues = {
  mode: "manual",
  name: "",
};
```

I tried removing it, but then React complained:

> Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: [https://reactjs.org/link/controlled-components](https://reactjs.org/link/controlled-components?utm_source=chatgpt.com)

If there’s a proper solution, I’d be more than happy to adopt it.